Tests run fine in 3.0.0.Alpha2 but fails in 3.0.0.Final

##### Issue Overview
The greeter test from the examples, testing simple injection using weld embedded
works fine with 3.0.0.Alpha2 but fails on 3.0.0.Final with java.lang.NoClassDefFoundError: jakarta/ejb/Singleton

##### Expected Behaviour
Tests to run without failing

##### Current Behaviour
Tests fail with 

    java.lang.NoClassDefFoundError: jakarta/ejb/Singleton
    at org.jboss.arquillian.container.weld.embedded.Utils.<clinit>(Utils.java:81)
    at org.jboss.arquillian.container.weld.embedded.WeldMockContainer.deploy(WeldMockContainer.java:106)
    ...
    Caused by: java.lang.ClassNotFoundException: jakarta.ejb.Singleton
    at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:636)

##### Steps To Reproduce
Using test:

    @ExtendWith(ArquillianExtension.class)
    public class GreeterTest {
        @Deployment
        public static JavaArchive createDeployment() {
            return ShrinkWrap.create(JavaArchive.class)
                .addClasses(
                    Greeter.class, 
                    PhraseBuilder.class)
                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
        }
        
        @Inject
        private Greeter greeter;
        
        @Test
        public void should_create_greeting() {
            assertThat(greeter.createGreeting("Earthling")).isEqualTo("Hello, Earthling!");
            greeter.greet(System.out, "Earthling");
        }
    }

With dependencies:

    <dependency>
      <groupId>org.jboss.arquillian</groupId>
      <artifactId>arquillian-bom</artifactId>
      <version>1.7.0.Alpha10</version>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
    <dependency>
      <groupId>org.jboss.arquillian.container</groupId>
      <artifactId>arquillian-weld-embedded</artifactId>
      <version>3.0.0.Final</version>
    </dependency>


##### Additional Information
I can resolve the issue by including the dependency:

    <dependency>
      <groupId>jakarta.ejb</groupId>
      <artifactId>jakarta.ejb-api</artifactId>
    </dependency>

But since it works without this in Alpha2, this feels like a bug.

Tested using JDK 16 and jakartaee-api 9.0.0

--- a/pom.xml
+++ b/pom.xml
@@ -102,7 +102,8 @@
                 <groupId>jakarta.ejb</groupId>
                 <artifactId>jakarta.ejb-api</artifactId>
                 <version>${ejb.api.version}</version>
-                <scope>provided</scope>
+                <!-- We need this dependency to be "compile" because in some scenarios, it doesn't need to be provided -->
+                <scope>compile</scope>
             </dependency>
             <dependency>
                 <groupId>jakarta.transaction</groupId>

